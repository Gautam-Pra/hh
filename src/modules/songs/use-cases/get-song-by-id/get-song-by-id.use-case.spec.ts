import { beforeAll, describe, expect, it } from 'vitest'
import { HTTPException } from 'hono/http-exception'
import { GetSongByIdUseCase } from '#modules/songs/use-cases'

describe('GetSongById', () => {
  let getSongById: GetSongByIdUseCase

  beforeAll(() => {
    getSongById = new GetSongByIdUseCase()
  })

  it('should return a song by id', async () => {
    const song = await getSongById.execute({ songIds: '3IoDK8qI' })

    expect(song[0]).toMatchSnapshot({
      playCount: expect.any(Number)
    })
  })

  it('should return a song by id and include lyrics', async () => {
    const song = await getSongById.execute({ songIds: 'L91uYhUm', includeLyrics: true })

    expect(song[0]).toMatchSnapshot({
      playCount: expect.any(Number),
      hasLyrics: expect.any(Boolean),
      lyricsId: song[0].lyricsId ? expect.any(String) : null
    })
  })

  it('should throw 404 error when song is not found', async () => {
    await expect(getSongById.execute({ songIds: 'invalid-id' })).rejects.toThrow(HTTPException)
  })
})
