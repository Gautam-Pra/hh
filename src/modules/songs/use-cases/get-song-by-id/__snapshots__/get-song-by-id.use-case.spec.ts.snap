// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GetSongById > should return a song by id 1`] = `
{
  "album": {
    "id": "23241654",
    "name": "Future Nostalgia",
    "url": "https://www.jiosaavn.com/album/future-nostalgia/ITIyo-GDr7A_",
  },
  "artists": {
    "all": [
      {
        "id": "573802",
        "image": [],
        "name": "Clarence Coffee Jr",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/clarence-coffee-jr-songs/2mX96rBxCYU_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
      {
        "id": "702498",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/587/Gypsy-Girl-A-Cappella--English-2017-20191123100913-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/587/Gypsy-Girl-A-Cappella--English-2017-20191123100913-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/587/Gypsy-Girl-A-Cappella--English-2017-20191123100913-500x500.jpg",
          },
        ],
        "name": "Sarah Hudson",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/sarah-hudson-songs/pW4Y,NKAPPI_",
      },
      {
        "id": "599533",
        "image": [],
        "name": "Stephen Kozmeniuk",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/stephen-kozmeniuk-songs/lS1-2YuC5oc_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "singer",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "primary_artists",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
  },
  "copyright": "â„— 2020  Dua Lipa Limited under exclusive license to Warner Records UK, a division of Warner Music UK Limited",
  "downloadUrl": [
    {
      "quality": "12kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_12.mp4",
    },
    {
      "quality": "48kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_48.mp4",
    },
    {
      "quality": "96kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_96.mp4",
    },
    {
      "quality": "160kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_160.mp4",
    },
    {
      "quality": "320kbps",
      "url": "https://aac.saavncdn.com/665/7790c3b9097592113008eaf1031d6e57_320.mp4",
    },
  ],
  "duration": 203,
  "explicitContent": true,
  "hasLyrics": false,
  "id": "3IoDK8qI",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/665/Future-Nostalgia-English-2020-20231110075246-500x500.jpg",
    },
  ],
  "label": "Warner Records",
  "language": "english",
  "lyricsId": null,
  "name": "Levitating",
  "playCount": Any<Number>,
  "releaseDate": "2020-03-27",
  "type": "song",
  "url": "https://www.jiosaavn.com/song/levitating/QyEEdT8IRno",
  "year": 2020,
}
`;

exports[`GetSongById > should return a song by id and include lyrics 1`] = `
{
  "album": {
    "id": "13192364",
    "name": "Skipping Stones",
    "url": "https://www.jiosaavn.com/album/skipping-stones/qVWQh1H3Lf4_",
  },
  "artists": {
    "all": [
      {
        "id": "599453",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-500x500.jpg",
          },
        ],
        "name": "Dan Reynolds",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dan-reynolds-songs/xoeLrr0aTEU_",
      },
      {
        "id": "561655",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-500x500.jpg",
          },
        ],
        "name": "Hans Zimmer",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/hans-zimmer-songs/Xhc-OKiRcRw_",
      },
      {
        "id": "2608037",
        "image": [],
        "name": "Timothy Randolph Edgar",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/timothy-randolph-edgar-songs/qHdEQDEI4zk_",
      },
      {
        "id": "599453",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-500x500.jpg",
          },
        ],
        "name": "Dan Reynolds",
        "role": "singer",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dan-reynolds-songs/xoeLrr0aTEU_",
      },
      {
        "id": "561655",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-500x500.jpg",
          },
        ],
        "name": "Hans Zimmer",
        "role": "singer",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/hans-zimmer-songs/Xhc-OKiRcRw_",
      },
      {
        "id": "599453",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-500x500.jpg",
          },
        ],
        "name": "Dan Reynolds",
        "role": "lyricist",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dan-reynolds-songs/xoeLrr0aTEU_",
      },
      {
        "id": "561655",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-500x500.jpg",
          },
        ],
        "name": "Hans Zimmer",
        "role": "lyricist",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/hans-zimmer-songs/Xhc-OKiRcRw_",
      },
      {
        "id": "2608037",
        "image": [],
        "name": "Timothy Randolph Edgar",
        "role": "lyricist",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/timothy-randolph-edgar-songs/qHdEQDEI4zk_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "599453",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-500x500.jpg",
          },
        ],
        "name": "Dan Reynolds",
        "role": "primary_artists",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dan-reynolds-songs/xoeLrr0aTEU_",
      },
      {
        "id": "561655",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/606/The-Dark-Knight-Rises-English-2012-20180423192208-500x500.jpg",
          },
        ],
        "name": "Hans Zimmer",
        "role": "primary_artists",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/hans-zimmer-songs/Xhc-OKiRcRw_",
      },
    ],
  },
  "copyright": "â„— 2018 KIDinaKORNER/Interscope Records",
  "downloadUrl": [
    {
      "quality": "12kbps",
      "url": "https://aac.saavncdn.com/048/5eee4e48523642a67affc80170d18bde_12.mp4",
    },
    {
      "quality": "48kbps",
      "url": "https://aac.saavncdn.com/048/5eee4e48523642a67affc80170d18bde_48.mp4",
    },
    {
      "quality": "96kbps",
      "url": "https://aac.saavncdn.com/048/5eee4e48523642a67affc80170d18bde_96.mp4",
    },
    {
      "quality": "160kbps",
      "url": "https://aac.saavncdn.com/048/5eee4e48523642a67affc80170d18bde_160.mp4",
    },
    {
      "quality": "320kbps",
      "url": "https://aac.saavncdn.com/048/5eee4e48523642a67affc80170d18bde_320.mp4",
    },
  ],
  "duration": 248,
  "explicitContent": false,
  "hasLyrics": Any<Boolean>,
  "id": "L91uYhUm",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/048/Skipping-Stones-English-2018-20180620230710-500x500.jpg",
    },
  ],
  "label": "Kid Ina Korner / Interscope",
  "language": "english",
  "lyrics": {
    "copyright": "Writer(s): Daniel Coulter Reynolds, Hans Florian Zimmer, Tim Randolph<br>Lyrics powered by www.musixmatch.com",
    "lyrics": "I wake up every morning with a new resolve<br>Two hands and one heart, let the world revolve<br>One eye to the ocean and one to the sky<br>One life of devotion and where am I?<br>One heart to love within my chest<br>One life to live<br><br>I keep skipping stones, hoping for a change<br>But things just stay the same<br>I keep walking roads, looking for a home<br>That I can call my own<br>Skipping stones<br>Iâ€²m skipping stones<br>I'm skipping stones<br>Iâ€²m skipping stones<br><br>One chance to reach out and find a place for me<br>One chance to find love for all humanity<br>One eye to the mountain and one to the sands<br>One dime for the fountain and one for my hand<br>One heart to love within my chest<br>One heart to love<br><br>I keep skipping stones, hoping for a change<br>But things just stay the same<br>I keep walking roads, looking for a home<br>That I can call my own<br>Skipping stones<br>I'm skipping stones<br>I'm skipping stones<br>Iâ€²m skipping stones<br><br>Skipping stones<br>Iâ€²m skipping stones<br>I'm skipping stones<br>Iâ€²m skipping stones<br><br>Love is not a product of reasonings and statistics<br>It just comes - none knows whence - and cannot explain itself<br>Love is a madness, if thwarted, it develops fast<br>When you fish for love, bait with your heart, not your brain<br><br>I keep skipping stones, hoping for a change<br>But things just stay the same<br>I keep walking roads, looking for a home<br>That I can call my own<br>Skipping stones<br>I'm skipping stones<br>Iâ€²m skipping stones<br>I'm skipping stones<br><br>Skipping stones<br>Iâ€²m skipping stones<br>I'm skipping stones<br>I'm skipping stones",
    "snippet": "I keep skipping stones, hoping for a change",
  },
  "lyricsId": null,
  "name": "Skipping Stones",
  "playCount": Any<Number>,
  "releaseDate": "2018-06-22",
  "type": "song",
  "url": "https://www.jiosaavn.com/song/skipping-stones/PFFaRC1YYl4",
  "year": 2018,
}
`;
